build_tools/benchmarks/comparisons/run_benchmarks.py

```diff
@@ -0,0 +1,143 @@
+    for benchmark in benchmarks:
+      results_array = [device_name, benchmark.model_name,
+                       benchmark.runtime, benchmark.driver, num_threads]
+      benchmark.num_threads = num_threads
+      results_array.extend(run_command(benchmark))
+      write_benchmark_result(results_array, results_path)
+
+
+def benchmark_desktop_gpu(device_name: str,
+    command_factories: list[BenchmarkCommandFactory], results_path: str):
+  benchmarks = []
+  for factory in command_factories:
+    benchmarks.extend(factory.generate_benchmark_commands("desktop", "gpu"))
+  for benchmark in benchmarks:
+    results_array = [device_name, benchmark.model_name,
+                     benchmark.runtime, benchmark.driver,
+                     benchmark.num_threads]
+    results_array.extend(run_command(benchmark))
+    write_benchmark_result(results_array, results_path)
+
+
+def benchmark_mobile_cpu(device_name: str,
+    command_factories: list[BenchmarkCommandFactory], results_path: str):
+  benchmarks = []
+  for factory in command_factories:
+    benchmarks.extend(factory.generate_benchmark_commands("mobile", "cpu"))
+
+  for _, tuple in enumerate(
+      [("80", 1), ("C0", 2), ("F0", 4), ("0F", 4), ("FF", 8)]):
+    taskset = tuple[0]
+    num_threads = tuple[1]
+    for benchmark in benchmarks:
+      results_array = [device_name, benchmark.model_name,
+                       benchmark.runtime, benchmark.driver, taskset,
+                       num_threads]
+      benchmark.taskset = taskset
+      benchmark.num_threads = num_threads
+      results_array.extend(run_command(benchmark))
+      write_benchmark_result(results_array, results_path)
+
+
+def benchmark_mobile_gpu(device_name: str,
+    command_factories: list[BenchmarkCommandFactory], results_path: str):
+  benchmarks = []
+  for factory in command_factories:
+    benchmarks.extend(factory.generate_benchmark_commands("mobile", "gpu"))
+
+  taskset = "80"
+  num_threads = 1
+  for benchmark in benchmarks:
+    results_array = [device_name, benchmark.model_name,
+                     benchmark.runtime, benchmark.driver, taskset, num_threads]
+    benchmark.taskset = taskset
+    benchmark.num_threads = num_threads
+    results_array.extend(run_command(benchmark))
+    write_benchmark_result(results_array, results_path)
+
+
+def main(args):
+  # Create factories for all models to be benchmarked.
+  command_factory = []
+  command_factory.append(MobilebertFP32CommandFactory(args.base_dir))
+
+  if args.mode == "desktop":
+    results_path = os.path.join(args.output_dir, "results.csv")
+    with open(results_path, "w") as f:
+      f.write(
+          "device,model,runtime,driver/delegate,threads,latency (ms),vmhwm (KB),vmrss (KB),rssfile (KB)\n")
+
+    if not args.disable_cpu:
+      benchmark_desktop_cpu(args.device_name, command_factory, results_path)
+    if not args.disable_gpu:
+      benchmark_desktop_gpu(args.device_name, command_factory, results_path)
+  else:
```

Comment:
Maybe add an assert, mostly for documentation

```suggestion
  else:
    assert(args.mode == "mobile")
```

Reply:
Thanks. Done.